src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -168,16 +170,23 @@ public void ObjectCreationOverloadResolution(ImmutableArray<MethodSymbol> constr
             RefKind returnRefKind = default,
             TypeSymbol returnType = null,
             bool isFunctionPointerResolution = false,
+            bool isExtensionMethodResolution = false,
             in CallingConventionInfo callingConventionInfo = default)
             where TMember : Symbol
         {
             var results = result.ResultsBuilder;
 
+            // No need to check for overridden or hidden methods if the members were
+            // resolved as extension methods and the extension methods are defined in
+            // types derived from System.Object.
+            bool checkOverriddenOrHidden = !(isExtensionMethodResolution &&
```

Comment:
nit: it feels like this might be easier to understand as `!isExtensionMethodResolution || !members.All(...)`

Reply:
To me, both forms seem equally difficult to read. I'll leave it as is since the current form matches the comment above which seems relatively clear.